apps/include/app_libctx.h

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef OSSL_APPS_LIBCTX_H
+# define OSSL_APPS_LIBCTX_H
+
+OSSL_LIB_CTX *app_create_libctx(void);
+OSSL_LIB_CTX *app_get0_libctx(void);
+int app_set_propq(const char *arg);
+const char *app_get0_propq(void);
+
+#endif
```

Comment:
I real benefit would be if the implementation was moved from `apps/lib/apps.c` to `apps/lib/app_libctx.c` while you're doing this.

Reply:
This wouldn't help.  `opt_printf_stderr()` would still be sucked in which pulls in apps.o

I don't remember the specifics of not wanting to `#include "apps.h"` in opt.c beyond knowing there was a reason.

Reply:
Because it is used in the tests, not just the apps (and apps.h should not get pulled into tests)

Reply:
I've moved them.